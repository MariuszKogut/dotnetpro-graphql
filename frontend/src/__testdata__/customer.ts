export const customerTestData = [
  { id: 1, name: 'Microsoft', location: 'USA' },
  { id: 2, name: 'Oracle', location: 'USA' },
  { id: 3, name: 'IBM', location: 'USA' },
]
