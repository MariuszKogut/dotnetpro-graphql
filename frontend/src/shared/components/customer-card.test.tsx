import React from 'react'
import { render } from '@testing-library/react'
import CustomerCard from './customer-card'
import { customerTestData } from '../../__testdata__/customer'

jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: jest.fn(),
    history: jest.fn(),
  }),
}))

describe('CustomerCard', () => {
  it('renders without errors', () => {
    const { container } = render(
      <CustomerCard customer={customerTestData[0]} />,
    )
    expect(container).toMatchSnapshot()
  })
})
