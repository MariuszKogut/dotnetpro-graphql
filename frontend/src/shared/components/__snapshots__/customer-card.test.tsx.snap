// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerCard renders without errors 1`] = `
<div>
  <div
    class="card card"
  >
    <div
      class="font-weight-bold card-header"
    >
      Microsoft
    </div>
    <div
      class="list-group list-group-flush"
    >
      <div
        class="list-group-item"
      >
        <span
          class="text-muted"
        >
          #
        </span>
         
        1
      </div>
      <div
        class="list-group-item"
      >
        <span
          class="text-muted"
        >
          Location
        </span>
         
        USA
      </div>
    </div>
    <div
      class="d-flex card-body"
    >
      <button
        class="btn btn-secondary"
        type="button"
      >
        Ã–ffnen
      </button>
    </div>
  </div>
</div>
`;
